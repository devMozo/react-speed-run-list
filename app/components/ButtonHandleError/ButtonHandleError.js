import './ButtonHandleError.scss';
import PropTypes from 'prop-types';

const ButtonHandleError = ({ onClick }) => (
  <div className="ButtonHandleError">
    <button onClick={onClick} type="button">
      {' '}
      Try Again
{' '}
    </button>
  </div>
);

ButtonHandleError.propTypes = {
  onClick: PropTypes.func,
};

export default ButtonHandleError;
