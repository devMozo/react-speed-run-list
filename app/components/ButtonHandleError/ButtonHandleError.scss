@import '../../fonts/index.scss';

.ButtonHandleError {
  width: 100%;
  text-align: center;
  margin: 20px 0;

  button {
    padding: 10px 20px;
    margin: 0 auto;
    background-color: #222;
    border: 0;
    color: #eee;
    font-family: $fontMedium;
    cursor: pointer;
    transition: 0.3s;

    &:hover {
      opacity: 0.6;
    }
  }
}
