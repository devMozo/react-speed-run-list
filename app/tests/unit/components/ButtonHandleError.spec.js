import React from 'react';
import { shallow } from 'enzyme';
import ButtonHandleError from '../../../components/ButtonHandleError/ButtonHandleError';

describe('<ButtonHandleError />', () => {
  const fnOnClick = jest.fn();

  it('should render correctly', () => {
    const component = shallow(<ButtonHandleError />);
    expect(component).toMatchSnapshot();
  });

  it('trigger click', () => {
    const component = shallow(<ButtonHandleError onClick={fnOnClick} />);
    component
      .find('button')
      .first()
      .simulate('click');
    expect(fnOnClick).toBeCalled();
  });
});
