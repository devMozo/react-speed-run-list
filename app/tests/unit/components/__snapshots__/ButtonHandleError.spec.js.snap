// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonHandleError /> should render correctly 1`] = `
<div
  className="ButtonHandleError"
>
  <button
    type="button"
  >
     
    Try Again
     
  </button>
</div>
`;
